src/auth/dto/authcreds.dto.ts

```diff
@@ -0,0 +1,15 @@
+import { IsString, IsNotEmpty, IsEmail } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class AuthCredsDto {
+  @IsString()
+  @ApiProperty({ description: 'The email of the user' })
+  @IsEmail()
```

Comment:
PullProofAI

There is a potential security vulnerability in the `AuthCredsDto` class. Storing passwords as plain strings is not secure. It's recommended to hash passwords before storing them in the database to enhance security.

To address this issue, consider using a hashing algorithm like bcrypt to securely store and compare passwords.

```diff
+  @IsString()
+  @ApiProperty({ description: 'The hashed password of the user' })
+  @IsNotEmpty()
+  password: string;
```


<!-- This is an auto-generated comment by PullProofAI -->